include/swift/AST/DiagnosticsSIL.def

```diff
@@ -71,6 +71,22 @@ ERROR(inout_argument_alias,none,
 NOTE(previous_inout_alias,none,
       "previous aliasing argument", ())
 
+ERROR(magic_identifier_missing_arg,none,
+      "missing argument for parameter %0 of literal expression %1",
+      (StringRef, StringRef))
+ERROR(magic_identifier_unexpected_arg,none,
+      "unexpected argument for parameter %0 of literal expression %1",
+      (StringRef, StringRef))
+ERROR(magic_identifier_invalid_arg_name,none,
+      "invalid argument name %0 passed to literal expression %1",
+      (StringRef, StringRef))
+ERROR(magic_identifier_unknown_arg_name,none,
+      "unknown argument name %0 passed to literal expression %1",
+      (StringRef, StringRef))
+WARNING(magic_identifier_poor_transcription,none,
```

Comment:
What would cause this to happen? Seems like it should probably be a full error, not just a warning.

Reply:
`magic_identifier_poor_transcription` you mean? If the argument is compiler-generated, then it won't have a source location to pull code from. Or, if you use the literal outside a default argument:

```swift
func f(arg: Int) {
  print(#transcription(of: arg)) // won't do what you want
}
```

Should it be a hard error? Maybe. I think that's a discussion we'd want to have.

Reply:
If I ask for the transcription of `arg` in the body of a function I fully expect to get the string "arg". Just as I'd expect `print(#transcription(of: arg + 1))` to print

```
arg + 1
```

Reply:
And that's the string you'd get, but it's probably not what a developer actually means. "arg" is not an interesting result.